### ISSUE TITLE
Figure out model for crossplane core vs extensions

### ISSUE THREAD

**Issue by jbw976** [crossplaneio/crossplane#531]
**Is this a bug report or feature request?**
* Feature Request

**What should the feature do:** The [extensions 1-pager](https://github.com/crossplaneio/crossplane/blob/master/design/extensions.md) described the core parts of crossplane that could be installed up front and suggested that all additional functionality could be installed as extensions.  We need to figure out where that line is and a plan for implementing it.

* Should each cloud provider be an extension?
* Should each resource type be an extension?
* Should each extension move to its own source code repository?

**Comment by hasheddan** [comment]
This is a very important line of distinction. It seems that if the user experience for installing extensions is ensured to be simple and painless then it would be good idea to move as many things as possible to extensions rather than core. This would allow for more control over what is installed, and though there is not a lot of overhead for having unused core pieces installed up front, it may be cleaner for an organization that doesn't use a particular cloud provider to not even have the ability to set it up without installing an extension.

**Comment by jbw976** [comment]
Related to #532

**Comment by negz** [comment]
I'm going to temporarily assign this issue to myself so I can braindump my thoughts on how we might approach this for infrastructure stacks. I'll release the issue once I've updated it.

**Comment by displague** [comment]
Related to #672 which disambiguates Infra stacks from app stacks.

**Comment by displague** [comment]
In Slack we fell into a naming conversation about stacks which also relates to the organization the stacks are expected to live in.

In terms of sort order on Github repo listing..
`microsoft/azure*` could be crowded but `microsoft/stack-*` would not.. 
`microsoft/crossplane-stack-azure` would not either..

On the Crossplane GH org, a cloud first naming convention could be hard to follow:
* `crossplane/azure-stack`
* `crossplane/something-else`
* `crossplane/google-stack`
* `crossplane/something-else`
* `crossplane/hetzner-stack`

Here are some examples from other projects I am aware of:
* `terraform-provider-{cloud}`
* `terraform-provisioner-{crossplane}`
* `terraform-{cloud}-{module}`
* `docker-machine-driver-{cloud}`
* `docker-volume-{cloud}`
* `fog-{cloud}`

But then you also find projects where the name was not well defined, so you get a proliferation of names..
* `{cloud}-{product}-csi-driver`
* `{cloud}/csi-driver`
* `{product}-csi-driver}`
* `csi-{cloud}`
As it is with Kubernetes CSI drivers, likewise for Kubernetes CCM drivers.

And you can also encounter organizations that would like to uphold their own naming convention on their repositories .. `{cloud-org}/{cloud}-{3rd-party-product}-{type-of-integration}`.  Some orgs would also prefer to keep a fork on their GH account, while there is not much to do about naming in this case, Issue tracking and responsibility can come into play.

Yet another aspect is the preferred GH org for integrations.  There are strong incentives to have terraform-providers live under the terraform-provider org (likewise for pulumi and fog).

Docker plugins and Terraform modules can live anywhere on the other hand, but should be registered in the terraform registry / docker plugin registry. Modules to be listed in the Terraform registry must follow a repository naming convention.

This may be fodder for the design around ~extensions~ stacks.
